src/lib/__tests__/date.test.ts

```diff
@@ -1,51 +1,271 @@
+import {
+  dateRange,
+  exhibitionStatus,
+  datesWithHours,
+  datesWithDaysAndHours,
+  singleDateWithoutHour,
+  singleDate,
+  singleHour,
+  formattedOpeningHours,
+  datesAreSameDay,
+  formattedHours,
+} from "lib/date"
 
-describe("date", () => {
-  describe("exhibitionPeriod", () => {
-    it("includes the start and end date", () => {
-      const period = exhibitionPeriod(
-        moment("2011-01-01T00:00-0400"),
-        moment("2014-04-19T00:00-0400")
+describe("date formatting", () => {
+  describe("formattedHours", () => {
+    it("If within same am/pm", () => {
+      const period = formattedHours(
+        "2022-12-30T08:00:00+00:00",
+        "2022-12-30T10:00:00+00:00"
+      )
+      expect(period).toBe("8:00 – 10:00am")
+    })
+
+    it("If within across both am/pm", () => {
+      const period = formattedHours(
+        "2021-12-05T08:00:00+00:00",
+        "2022-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe("8:00am – 5:00pm")
+    })
+  })
+
+  describe("datesAreSameDay", () => {
+    it("Dates are the same day", () => {
+      const period = datesAreSameDay(
+        "2022-12-30T20:00:00+00:00",
+        "2022-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe(true)
+    })
+
+    it("Dates are not the same day", () => {
+      const period = datesAreSameDay(
+        "2021-12-05T20:00:00+00:00",
+        "2022-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe(false)
+    })
+  })
+
+  describe("formattedOpeningHours", () => {
+    const realNow = Date.now
+    beforeEach(() => {
+      Date.now = () => new Date("2018-01-30T03:24:00") as any
+    })
+    afterEach(() => {
+      Date.now = realNow
+    })
+
+    it("Event opens in the future", () => {
+      const period = formattedOpeningHours(
+        "2021-12-05T20:00:00+00:00",
+        "2022-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe("Opens Dec 5 at 8pm")
+    })
+
+    it("Event is running and closes in the future", () => {
+      const period = formattedOpeningHours(
+        "2017-12-05T20:00:00+00:00",
+        "2019-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe("Closes Dec 30 at 5pm")
+    })
+
+    it("Event ended in the past and is closed", () => {
+      const period = formattedOpeningHours(
+        "2016-12-05T20:00:00+00:00",
+        "2016-12-30T17:00:00+00:00"
       )
+      expect(period).toBe("Closed")
+    })
+  })
+
+  describe("singleHour", () => {
+    it("includes single date in this year", () => {
+      const period = singleHour("2018-12-05T20:00:00+00:00")
+      expect(period).toBe("8:00pm")
+    })
+  })
+
+  describe("singleDate", () => {
+    const realNow = Date.now
+    beforeEach(() => {
+      Date.now = () => new Date("2018-01-30T03:24:00") as any
+    })
+    afterEach(() => {
+      Date.now = realNow
+    })
+
+    it("includes single date in this year", () => {
+      const period = singleDate("2018-12-05T20:00:00+00:00")
+      expect(period).toBe("Dec 5 at 8:00pm")
```

Comment:
Missing the TZ component. https://www.notion.so/artsy/Formatting-Time-f6f4aaeeba414583a207013fec5a9794#a9f31923b35d41579c400bd76ce3ba13